controllers/v1/observationSubmissionsController.js

```diff
@@ -221,17 +223,34 @@ module.exports = class ObservationSubmissions extends Abstract {
       }
       if( observationDocument.userRoleInformation && Object.keys(observationDocument.userRoleInformation).length > 0 ){
           submissionDocument.userRoleInformation = observationDocument.userRoleInformation;
-      } else if( req.body && req.body.role && !observationDocument.userRoleInformation ){
-          submissionDocument.userRoleInformation = req.body;
+      } else if( !observationDocument.userRoleInformation ){
+        let userRoleAndProfileInformation = {};
+        let userProfile = await sunbirdUserProfile.profile( req.userDetails.userToken, req.userDetails.userId );
+        
+        if (!userProfile.success || 
+            !userProfile.data ||
+            !userProfile.data.response
+        ) {
+            userRoleAndProfileInformation = {};
+            if ( req.body && req.body.role ) {
+              userRoleAndProfileInformation = req.body;
+            }
+        } else {
+            let userProfileDoc = userProfile.data.response;
+            let userProfileDetails = await observationsHelper.formatProfileData( userProfileDoc );
```

Comment:
@VISHNUDAS-tunerlabs remove this function from observation helper make it in utils